### PR TITLE
Enhanced Game Restart Logic and GUI Introduction

### PR DESCRIPTION
**Pull request by Leviaria**
This update introduces several key improvements to ClashRoyaleBuildABot

**1. Enhanced Game Restart Logic:**

- The bot now intelligently handles scenarios where reward screens or menus might appear after a match, waiting for a specific duration before attempting to detect the "Battle" button.
- If the button remains undetected, the game is automatically restarted using ADB commands to ensure uninterrupted gameplay.

**2. New GUI (Graphical User Interface):**
- A new GUI has been added to the bot, optimized for Windows 11 terminals.
- The GUI functions smoothly on Windows 10 as well, with a few minor visual differences.
- Settings customization and further optimizations are planned for future updates.
- The GUI provides improved text clarity, and in DEBUG mode, previously hidden information can be viewed without impacting performance.

GUI
image

Have Fun! :)

### CODE REVIEW

File: example/custom_bot.py [hunk 1]
```diff
@@ -50,63 +28,50 @@ def _preprocess(self):
...
 
-            # End of game detected, but no actions performed yet
             if self.state['screen'] == 'end_of_game':
-                self._click(*SCREEN_CONFIG['end_of_game']['click_coordinates'])
-                print("End of game detected, waiting for 10 seconds...")
-                self.pause_until = time.time() + 10  # Set pause for 10 seconds
-                self.end_of_game_clicked = True  # Mark the end of the game
-                time.sleep(2)  # Additional pause to ensure state is updated
-                continue  # Skip to the next loop iteration without performing actions
+                print(f"[{datetime.now().strftime('%H:%M:%S')}] [INFO] End of game detected. Waiting 10 seconds for battle button...")
```
Comment (Pbatch) [inline example/custom_bot.py:54]: We can use a library like Loguru to do this logging easier. I'll update it in a future MR.

**Comment by Pbatch** [comment]
This is awesome man. Great work :D 

I'm tempted to go back and try and fix the various machine learning components now, I haven't looked at the code properly for a very long time.

**Comment by Pbatch** [comment]
When this is done I'll make sure the demo video makes it to the README

File: example/custom_bot.py (hunk 1 above)
Comment (Leviaria) [reply example/custom_bot.py:54]: Definitely open to suggestions for improvements, I'll look into Loguru after work tomorrow. I think this project will bring me some fun days in the future. :)

**Comment by Leviaria** [comment]
**Enhance Logging with Loguru Integration**

This 3 changes replaces the basic print statements with Loguru. 

This enhancement provides:

- Structured Logging: Logs are organized with timestamps, levels (e.g., INFO, DEBUG, ERROR), and clear messages for easier debugging and analysis.

- Customizable Log Files: Log files are now saved in the clashroyalebuildabot/logs directory with timestamps, making it easier to track bot activity over time.

Additional optimizations are planned for this evening to further streamline the code while maintaining its current efficiency.

image

**Comment by Pbatch** [comment]
I've separated your code into `windows_main.py` so that Mac/Linux users still have something to run.

@all-contributors please thank @Leviaria for code, examples and tutorials.

**Comment by Pbatch** [comment]
I've separated your code into windows_main.py so that Mac/Linux users still have something to run.

https://github.com/all-contributors please add @Leviaria for code, examples and tutorials.

**Comment by Pbatch** [comment]
I've separated your code into windows_main.py so that Mac/Linux users still have something to run.

@all-contributors please add @Leviaria for code, examples and tutorials.

**Pull request closed without merging**